### DISCUSSION TITLE
Yoink `slurmhelpers` out of the Slurm snap

### DISCUSSION THREAD

**Discussion by NucciTheBoss** [canonical/hpc-team#3] [General]
As the title suggests, I want to drop `slurmhelpers` from the Slurm snap package. What originally started out as "hey this is pretty cool" has proved to be more trouble than it's worth. There's a couple good reasons to consider dropping it:

- `slurmhelpers` is blocking our migration to `core24` which forces us to still rely on the rocm PPA we have set up for Jammy Jellyfish
  - https://github.com/charmed-hpc/slurm-snap/issues/19
- It's holding up CI for HPC libs
  - https://github.com/charmed-hpc/slurm-snap/issues/47
- Synchronization issues
  - The _slurm.conf_ configuration that exists within the snap's data may not always be in sync with the actual _slurm.conf_ file under `$SNAP_COMMON`. You can edit the _slurm.conf_ file directly and those changes will not be reflected within the snaps configuration.

### Alternatives

We still need the `.env` file to help configure how the Slurm and MUNGE daemons within the Slurm snap package behave. For this we're using the `dotenv` Python package. Tbh I would prefer if we just removed Python entirely, and replaced it with something else. Our options are:

1. Use a dotenv CLI utility from the Ubuntu Archive and use shell hooks.
2. Can use something quickly written in Go that leverages the [`godotenv`](https://github.com/joho/godotenv) package.

Either way I think we should aim to remove slurmhelpers so it removes current and doesn't cause any future blockers. Would like to know your thoughts @jedel1043, @wolsen, and @jamesbeedy :thinking:

**Comment by jedel1043** [comment]
Yeah, that sounds like a good idea. The alternatives are a lot more portable overall, and switching to a more portable option will definitely make the snap a lot more robust.

**Comment by jamesbeedy** [comment]
@NucciTheBoss I like the idea of removing it from the snap and putting it into the charm and useing it for generic `slurm.conf` generation for any `slurm.conf`, not just for the snapped slurm. +1

**Comment by NucciTheBoss** [comment]
Seems like we're in agreement here so I will begin the yoinking process :hammer_and_wrench:

**Comment by NucciTheBoss** [comment]
`slurmhelpers` has been removed from the Slurm snap :wrench:
